test/IIS.StaticShim.FunctionalTests/StaticShimTests.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Server.IIS.FunctionalTests.Utilities;
+using Microsoft.AspNetCore.Server.IntegrationTesting;
+using Microsoft.AspNetCore.Testing.xunit;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Server.IISIntegration.FunctionalTests
+{
+    [Collection(PublishedSitesCollection.Name)]
+    public class StaticShimTests : IISFunctionalTestBase
+    {
+        private readonly PublishedSitesFixture _fixture;
+
+        public StaticShimTests(PublishedSitesFixture fixture)
+        {
+            _fixture = fixture;
+        }
+
+        [ConditionalFact]
+        public async Task CheckStaticShimIsUsed()
+        {
+            var iisDeploymentParameters = _fixture.GetBaseDeploymentParameters(publish: true);
+            iisDeploymentParameters.PreservePublishedApplicationForDebugging = true;
+            var deploymentResult = await DeployAsync(iisDeploymentParameters);
+
+            var arch = iisDeploymentParameters.RuntimeArchitecture == RuntimeArchitecture.x64 ? $@"x64\aspnetcorev2.dll" : $@"x86\aspnetcorev2.dll";
+
+            var fileInfo = FileVersionInfo.GetVersionInfo(Path.Combine(AppContext.BaseDirectory, arch));
+            var response = await deploymentResult.HttpClient.GetAsync("/HelloWorld");
+
+            StopServer();
+
+            // File version is sent to debug log (hard to read win32 dll file version in c#)
+            Assert.Contains(TestSink.Writes, context => context.Message.Contains(fileInfo.FileVersion));
```

Comment:
Can you use IISTestSiteFixture and look at loaded modules of the process?

Reply:
I'm assuming you are referring to the HostProcess on IISTestSiteFixture?

Reply:
Yep